addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -674,7 +674,30 @@ def write(self, values):
         employee_id = values.get('employee_id', False)
         if values.get('state'):
             self._check_approval_update(values['state'])
-        result = super(HolidaysAllocation, self).write(values)
+
+        if 'number_of_days_display' in values or 'number_of_hours_display' in values:
```

Comment:
Maybe use the opposite condition so that less code is indented:
```suggestion
        if 'number_of_days_display' not in values and 'number_of_hours_display' not in values:
            self.add_follower(employee_id)
            return super().write(values)
        previous_consumed_leaves = ...
```